GH-597: second review

bertllll opened the pull request.

bertllll committed:
GH-595: better design of NotifyLaterHandle and NotifyHandle (#121)

* GH-595: first, probably wrong version

* GH-595: finishing the new design: arrangement and cleaness

* GH-595: adding where clauses to traits

* GH-595: fixes after a review

* GH-595: enhancing contract integration tests

Co-authored-by: Bert <Bert@Bert.com>

FinsaasGH committed:
Finsaas gh cimatrix (#124)

* Update ci-matrix.yml

* Create s3-release.yml

* Update ci-matrix.yml

* Update ci-matrix.yml

* Update s3-release.yml

FinsaasGH committed:
feat: update ci-matrix.yml and s3-release.yml to new pipeline (#127)

* fix: update branches in ci-matrix,yml

* Update ci-matrix.yml

* Update ci-matrix.yml

* Update ci-matrix.yml

* Update ci-matrix.yml

* Update s3-release.yml

* Update s3-release.yml

* Update s3-release.yml

* Update s3-release.yml

* Update s3-release.yml

* Update ci-matrix.yml

* Update s3-release.yml

* Update s3-release.yml

* Update ci-matrix.yml

* Update s3-release.yml

* Update ci-matrix.yml

* Update ci-matrix.yml

* Update s3-release.yml

* Update s3-release.yml

* Update ci-matrix.yml

bertllll committed:
GH-349: broadcasting to UIs about events through logs (#122)

* GH-349-initial-sketch: sketched for Utkarsh

* GH-349: add broadcast handle for UiLogBroadcast

* GH-349: modify the logger.rs

* GH-349-new: interim commit; trying to find a platform for fine testing of the new messaging

* GH-349-new: proposal of solution with otlined steps; overloading test revitalized and made more stable

* GH-349: feat - write tests for type conversions of SerializableLogLevel

* feat - refactor generic_log and write tests for all four possibilities

* GH-349: reafctor - improvements in logger.rs

* GH-349-new: test example of one of the variants in the big match statement; supposed to serve to Utkarsh

* GH-349: feat - enhance tests for generic logs in logger.rs

* GH-349-new: refactor - remove comments guided for writing tests

* GH-349-new: functions and tests under logger.rs likely complete

* GH-349-new: all tests in logger and actor system factory probably ready; before experiment with upgrading initi_test_loggigng() to prevent sending actor messages

* GH-349-new: unsafe removed

* GH-349-new: integration tests finished; formatting

* GH-349-new: auto review

* GH-349: bad assumption with release and feature argument

* GH-349: added tech description and changed build.sh script back

* GH-349: first try on three platforms with new integration test for dumping dead clients

* GH-349: a little fix for Windows in a troublesome integration test, probably ain't working

* GH-349: another attempt with that integration test; also a special unit test in websocketsupervisor

* GH-349: just a try to see a specific test in Actions, to do that I need to ignore a different test that causes troubles there for Win

* GH-349: promissing repairs for an integration test and aslo a unit test; both troublesome

* GH-349: ignoring a problematic test again

* GH-349: trying unignore the test again, maybe a race - I want to see it; also modifying the integration test

* GH-349: integration test should work now; actions try

* GH-349: build of ClientBuilder and Client Wrappers are ready for use

* GH-349: last modifications before tests, done; finished verify_tools parts and enhanced formatting for broadcasts shown in CLI

* GH-349: auto-review

* GH-349: part of refactoring; I just realized I stashed some code before

* GH-349: finishing repairment

* GH-349: auto-review fixes

* GH-349: review two answered

Co-authored-by: utkarshg6 <utkarshabhinav.torres@gmail.com>
Co-authored-by: Bert <Bert@Bert.com>

kauri-hero committed:
update Practice with reference to version & QA

kauri-hero committed:
update practice with version release info

kauri-hero committed:
DRAFT: Update docs/branch (#133)

* Update node top level README

add list of component README files for easy navigation

* Added reference to UI-Interface-Interface

Added reference to UI-Interface-Interface document that lives in the Node repo main folder

* Update README.md

* Delete SHRD-faucet

This is no longer usable and is not connected to the MASQ token contracts

* Update to blockchain service details 

Update to blockchain service details about Polygon, testnets, Alchemy 
also removed old images

* wording and formattting

* update LICENSE

updated with corrected copyright for MASQ, and added info regarding previous codebase being removed from GitHub and original repo disappeared

* Update Practice with QA blurb

Thanks Stew

kauri-hero committed:
adjust copyright

kauri-hero committed:
remove old DNS doc

kauri-hero committed:
remove old DNS doc

kauri-hero committed:
remove old DNS doc

kauri-hero committed:
delete old faucet info

kauri-hero committed:
updated with component readmes

kauri-hero committed:
remove old logo

kauri-hero committed:
remove old logo2

kauri-hero committed:
Create brand-guidelines.md

kauri-hero committed:
add masq logos

kauri-hero committed:
remove to move logos

kauri-hero committed:
add masq logos

kauri-hero committed:
added masq logo to readme

Commit:
GH-597: backup; closely before finishing / fixed an old issue with annoying 'Send' requirement; introducet a new concept of an arbitrary_id_stamp for testing specific situations

Commit:
GH-597: finished the concept of ArbitraryIdStamp

Commit:
GH-597: merging in master

Commit:
GH-597: updated github yml files; the release one is now more optimized

Commit:
GH-597: last polishing before another review

dnwiebe commented on node/src/bootstrapper.rs:
```diff
@@ -1689,10 +1722,12 @@ mod tests {
-            ListenerHandlerNull::new(vec![third_message]).bind_port_result(Ok(()));
+        let one_listener_handler = ListenerHandlerNull::new(vec![first_message, second_message])
+            .bind_port_result(Ok(()))
+            .stop_runtime_after_prepared_messages_exhausted();
```
You suggested "stop_polling..." instead of "stop_runtime..."

dnwiebe commented on node/src/bootstrapper.rs:
```diff
@@ -2080,19 +2113,26 @@ mod tests {
-            _actor_factory: Box<dyn ActorFactory>,
-            _persist_config: &mut dyn PersistentConfiguration,
+            actor_factory: Box<dyn ActorFactory>,
+            persist_config: Box<dyn PersistentConfiguration>, //the trait cannot be sent safely
```
Can erase the comment

dnwiebe commented on node/src/test_utils/mod.rs:
```diff
@@ -790,6 +790,36 @@ pub mod unshared_test_utils {
+    //It actually is handy for a very specific scenario:
+    //
+    //Consider a test definition area. We initiate a mocked trait object "O" encapsulated in a Box (so we will be
+    //moving ownership) and we plan to past it in a function A. The function contains other functions like
```
"paste" instead of "past"

dnwiebe commented on node/src/test_utils/mod.rs:
```diff
@@ -790,6 +790,36 @@ pub mod unshared_test_utils {
+    //will be always different than what we grab in C, because a move occurred in between, by moving the Box around.
+    //Downcasting is also a pain and not proving anything alone.
+    //
+    //That's why we can add a test-only method to our arbitrary trait by this macro. It allows implement
```
"allows implementing" instead of "allows implement."

dnwiebe commented on node/src/test_utils/mod.rs:
```diff
@@ -790,6 +790,36 @@ pub mod unshared_test_utils {
+    //Downcasting is also a pain and not proving anything alone.
+    //
+    //That's why we can add a test-only method to our arbitrary trait by this macro. It allows implement
+    //a method fetching a made up id which we initially gives the object at the start of the test. It can
```
"give," not "gives"

dnwiebe commented on node/src/actor_system_factory.rs:
```diff
@@ -1776,8 +1776,10 @@ mod tests {
-            Box::new(PersistentConfigurationMock::default().set_arbitrary_id_stamp(4578112));
+        let persistent_config = Box::new(
+            PersistentConfigurationMock::default()
+                .set_arbitrary_id_stamp(ArbitraryIdStamp::new(457812)),
```
It might make sense to have ArbitraryIdStamp be opaque. You'd create one with ::new(), and you'd have no idea what value it contained; the value could be randomized or sequential or calculated from a timestamp or whatever--just not supplied by the user.  It'd be Copy, so you'd hand it into .set_arbitrary_id_stamp() and retain it, and then assert it later.

Commit:
GH-597: second review addressed

dnwiebe committed:
GH 526 Stop Crediting Debtors Multiple Times (#117)

* GH-526: Now acquiring new start block; not saving it yet

* GH-526: A few more tests.

* GH-526: Mostly removing PersistentConfiguration from Accountant

* Blockchain client server needs some help

* GH-526: First run at custom-server MBCS is working

* GH-526: MBCS is working

* GH-526: Added --scans and scan command

* GH-526 Interim commit

* GH-526: Interim commit

* GH-526: Slowly writing multinode test

* Added --scans to multinode; a little progress on the test

* Blockchain client mock server and UI client for multinode tests

* Need to merge in master

* GH-526: master is merged, but tests fail

* GH-526: Master merged; all preexisting tests passing

* GH-526: Accountant now handles ResponseSkeleton properly; BlockchainBridge to follow

* GH-526: BlockchainBridge now turns around the ResponseSkeletons properly

* GH-526: Interim commit

* GH-526: Multinode test is working; code needs help

* BlockchainBridge now sending ScanError messages back to Accountant

* GH-526: Code is better; now why is the test failing?

* GH-526: Scan errors are making it back to masq now

* Added THINGS_TO_DO.md

* GH-526: Compile errors in accountant/mod.rs down to five

* GH-526: Compile errors gone from accountant/mod.rs

* GH-526: Tests in accountant/mod.rs passed

* GH-526: All unit tests passing

* GH-526: New multinode test ignored, all (non-ignored) tests passing

* GH-526: Pre-review issues

* Formatting

* Increased a timeout. Effect on Actions?

* GH-526: Delayed masq integration test until after GH-438.

* GH-526: Formatting

* Review issues

* Review issues

* Review issues

* Node tests pass

* Formatting

* No longer updating start_block incorrectly in the receivable_dao

* interim merge for GH-597 (#128)

* GH-597: code and tests established; some new features or work style used: no_share_test wider use

* GH-597: little corrections and formatting

* GH-597: auto review

* GH-597: cosmetics, added a few places with logs

* GH-597: A rookie mistake :) not running and so forgetting integration tests

Co-authored-by: Bert <Bert@Bert.com>

* Revert "Merge branch 'interim-test-070522' into GH-526"

This reverts commit 550df21bdbba975a39f42f935ded97f99bad8acf, reversing
changes made to f99391fb57d222feec5c8c778d75f9ae40c941e1.

* Revert

* Modified Cargo.tomls to add version 0.6.0

* GH-526: Review issues

* GH-526: review error remedy

Co-authored-by: KauriHero <55721017+kauri-hero@users.noreply.github.com>
Co-authored-by: Bert <Bert@Bert.com>

kauri-hero committed:
update from reviews

kauri-hero committed:
update and copyright bump

kauri-hero committed:
copyright bump [skip ci]

[skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

kauri-hero committed:
copyright bump [skip ci]

bertllll committed:
Merge branch 'master' into docs/contributing-plus-cleanup

Commit:
GH-597: mergin master in

Commit:
Merge branch 'docs/contributing-plus-cleanup' into GH-597

Commit:
docs/contributing-plus-cleanup: last polishment

Commit:
Merge branch 'docs/contributing-plus-cleanup' into GH-597

FinsaasGH committed:
Docs/contributing plus cleanup (#138)

* update Practice with reference to version & QA

* update practice with version release info

* DRAFT: Update docs/branch (#133)

* Update node top level README

add list of component README files for easy navigation

* Added reference to UI-Interface-Interface

Added reference to UI-Interface-Interface document that lives in the Node repo main folder

* Update README.md

* Delete SHRD-faucet

This is no longer usable and is not connected to the MASQ token contracts

* Update to blockchain service details 

Update to blockchain service details about Polygon, testnets, Alchemy 
also removed old images

* wording and formattting

* update LICENSE

updated with corrected copyright for MASQ, and added info regarding previous codebase being removed from GitHub and original repo disappeared

* Update Practice with QA blurb

Thanks Stew

* adjust copyright

* remove old DNS doc

* remove old DNS doc

* remove old DNS doc

* delete old faucet info

* updated with component readmes

* remove old logo

* remove old logo2

* Create brand-guidelines.md

* add masq logos

* remove to move logos

* add masq logos

* added masq logo to readme

* update from reviews

* update and copyright bump

* copyright bump [skip ci]

[skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* copyright bump [skip ci]

* docs/contributing-plus-cleanup: last polishment

Co-authored-by: KauriHero <55721017+kauri-hero@users.noreply.github.com>
Co-authored-by: Bert <65427484+bertllll@users.noreply.github.com>
Co-authored-by: Bert <Bert@Bert.com>

bertllll committed:
Merge branch 'master' into GH-597

Commit:
GH-597: a little correction in docs

Commit:
Merge branch 'GH-597' of github.com:MASQ-Project/Node into GH-597

Commit:
GH-597: fixed a labile test in WebsocketSuperwisor and added an explanation of an unclear solution around there too

Commit:
GH-597: removing unknown extern crate usage

Commit:
GH-597: trying to get the release build for Node working properly

Commit:
GH-597: removing an optimized build script because of an issue on Mac; we don't know how to fix that at the moment
